### PR TITLE
Add Documenter to build documentation

### PR DESCRIPTION
**Pull request by TestSubjector**
* Added basic Documenter Setup
* Ported parts of the documentation

TODO
- [x]  Port all the existing documentation to Documenter

### CODE REVIEW

File: src/fits.jl [hunk 1]
```diff
@@ -122,7 +122,20 @@ function getindex(f::FITS, name::AbstractString, ver::Int=0)
     return f.hdus[i]
 end
 
+"""
+    close(f::FITS)
 
+### Purpose ###
+
+Close the file. Subsequent attempts to operate on `f will result in an error.
```
Comment (giordano) [inline src/fits.jl]: Missing closing backtick after f

File: src/header.jl [hunk 2]
```diff
@@ -193,6 +193,21 @@ end
...
 # Public API
 
+
+"""
+    read_key(hdu, key)
+
+Read the specified key and return a tuple of `(value, comment)`.
+
+The key, can be either the index of the header record (Integer)
+or the header keyword (ASCIIString).
```
Comment (giordano) [inline src/header.jl]: `ASCIIString` has been removed in julia 0.5, use `String` instead. There are also other uses of `ASCIIString`. See also #75.

File: src/libcfitsio.jl [hunk 3]
```diff
@@ -862,6 +1067,28 @@ function fits_read_col{T}(f::FITSFile,
     fits_assert_ok(status[])
 end
 
+"""
+    fits_write_col(f, colnum, firstrow, firstelem, data)
+
+Write some data in one column of a ASCII/binary table.
+
+If there is no room for the elements, new rows will be created. (It is therefore
+useless to call :func:`fits_insert_rows` if you only need to *append* elements
```
Comment (kbarbary) [inline src/libcfitsio.jl]: Convert `:func:` to `@ref`.

**Comment by kbarbary** [comment]
Looking good so far! I guess me or @giordano will have to set up travis ssh keys, but perhaps that's better done after this PR is merged?

**Comment by giordano** [comment]
There is no rush to do it right now, but I'd do that before merging the PR, so that merge commit readily triggers building of the docs.

**Comment by kbarbary** [comment]
OK, I've set up the GitHub/Travis ssh keys for Documenter.

**Comment by kbarbary** [comment]
How's this going @TestSubjector? If you're short on time, we could do without adding the examples sections for the purpose of this PR.

**Comment by TestSubjector** [comment]
Yeah, this way will work much better for me. I'll work on and make a new PR for the examples documentation, when things become a little less hectic on my side.

**Comment by kbarbary** [comment]
I guess this is ready to merge?

**Comment by TestSubjector** [comment]
All clear from my side 🏁

**Comment by kbarbary** [comment]
Looks like the docs build didn't succeed on Travis. I think we need a `mkdocs.yml` file: https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#The-MkDocs-mkdocs.yml-File-1
and also arguments `deps` and `julia = "0.6"` in the `deploydocs()` function in `docs/make.jl`.

**Comment by giordano** [comment]
Travis on GNU/Linux with nighly Julia is successful, maybe the problem is that the [`src/index.md`](https://github.com/JuliaAstro/FITSIO.jl/blob/c9d5726395781f8c55baee4a7a21e8349012749c/docs/src/index.md) file is actually empty?

**Comment by kbarbary** [comment]
Oh, that's probably not good either.

The travis job can succeed while the doc build fails, because docs are built `after_success`. Looking at the log here: https://travis-ci.org/JuliaAstro/FITSIO.jl/jobs/269730603#L3836

**Pull request merged**